### PR TITLE
textDocument/implementation

### PR DESCRIPTION
**Pull request by Cosson2017**
add textDocument/implementation

### CODE REVIEW

File: autoload/lsp/capabilities.vim [hunk 1]
```diff
@@ -35,6 +35,10 @@ function! lsp#capabilities#has_workspace_symbol_provider(server_name) abort
     return s:has_bool_provider(a:server_name, 'workspaceSymbolProvider')
 endfunction
 
+function! lsp#capabilities#has_implementation_provider(server_name) abort
+    return s:has_bool_provider(a:server_name, 'implementationProvider')
```
Comment (prabirshrestha) [inline autoload/lsp/capabilities.vim]: spec says `implementationProvider?: boolean | (TextDocumentRegistrationOptions & StaticRegistrationOptions);` but for now seems ok. we can support it in the future if there is an ask.

File: plugin/lsp.vim [hunk 2]
```diff
@@ -31,3 +31,4 @@ command! LspRename call lsp#ui#vim#rename()
 command! LspWorkspaceSymbol call lsp#ui#vim#workspace_symbol()
 command! LspDocumentFormat call lsp#ui#vim#document_format()
 command! -range LspDocumentRangeFormat call lsp#ui#vim#document_range_format()
+command! LspImplementation call lsp#ui#vim#implementation()
```
Comment (prabirshrestha) [inline plugin/lsp.vim]: missing documentation

**Comment by Cosson2017** [comment]
Ok,I will add later

File: README.md [hunk 3]
```diff
@@ -60,6 +60,7 @@ Refer to docs on configuring omnifunc or [asyncomplete.vim](https://github.com/p
 |`:LspReferences`| Find references |
 |`:LspRename`| Rename symbol |
 |`:LspWorkspaceSymbol`| Search/Show workspace symbol |
+|`:LspImplementation` | Show implementation of interface |
```
Comment (prabirshrestha) [inline README.md]: readme should be in alphabetical order so it is easier for users to find it.

**Comment by Cosson2017** [comment]
Fixed

**Comment by prabirshrestha** [comment]
@Cosson2017 we should also add it in the doc/vim-lsp.txt

**Comment by prabirshrestha** [comment]
@Cosson2017 thanks merged.

**Pull request merged**